### PR TITLE
Enable unit test debug with delve

### PR DESCRIPTION
**Pull request by soltysh**
@stevekuznetsov ptal

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
I'd rather call the variable something like `$DELVE` since `$DEBUG` isn't very specific.

Internalize the variable on line ~78 ish like the others.

Add the doc for this var to the comment at the top of the script.

File: hack/test-go.sh [hunk 1]
```diff
@@ -288,6 +288,9 @@ elif [[ -n "${coverage_output_dir}" ]]; then
     # we can clean up all of the coverage reports at once as they all exist in subdirectories of ${coverage_output_dir}/${OS_GO_PACKAGE}
     # and they are the only files found in those subdirectories
     rm -rf "${coverage_output_dir:?}/${OS_GO_PACKAGE}"
+
+elif [[ -n "${DEBUG-}" ]]; then
```
Comment (stevekuznetsov) [inline hack/test-go.sh]: Unless we don't want the 'set-but-null' behavior, let's stick to the more common defaulting substitution for uniformity

```
+elif [[ -n "${DEBUG:-}" ]]; then
                    ^
```

File: hack/test-go.sh [hunk 2]
```diff
@@ -288,6 +288,9 @@ elif [[ -n "${coverage_output_dir}" ]]; then
...
+    dlv test ${test_packages}
```
Comment (stevekuznetsov) [inline hack/test-go.sh]: Add a comment here to explain this branch of the `elif`.

Comment (stevekuznetsov) [inline hack/test-go.sh]: Do we also want to have a binary check for `dlv` and spit out our go-to 'binary is missing' message here if we don't find it?

**Comment by soltysh** [comment]
I've changed the envvar to `DLV_DEBUG` including the existing places, so it's the same everywhere. ptal

**Comment by soltysh** [comment]
doc comment added, merging upon approval

[merge]

**Comment by soltysh** [comment]
@stevekuznetsov any chances to document other scripts similarly to how `hack/test-go.sh` is?

**Comment by soltysh** [comment]
Flake #9624

re-[merge]

**Comment by stevekuznetsov** [comment]
@soltysh we can dream :)

**Comment by soltysh** [comment]
#9624 again

re-[merge]

**Comment by marun** [comment]
For future please tag changes affecting the networking team with @openshift/networking so we have a chance to review before merge.  

A couple of references to NETWORKING_DEBUG were missed.  I've posted #9656 to remedy.

**Comment by marun** [comment]
@soltysh I tried using DLV_DEBUG with hack/test-go.sh and saw the following error: 

     could not launch process: fork/exec ./debug.test: operation not permitted

Any thoughts?

**Comment by soltysh** [comment]
Can you give me the exact line you run? I didn't have any problems when running:`DLV_DEBUG=y hack/test-go.sh` or additionally passing a specific subdir for tests.

**Comment by marun** [comment]
@soltysh I was seeing the problem running `DLV_DEBUG=y hack/test-go.sh` on centos 7.2, both in an f23 container (golang 1.5.x) and directly on the host (golang 1.4.x).  It works fine when I run the same command in an f23 vm, so I guess the problem is with my host.

**Pull request merged**